group_project_v2/group_project.py

```diff
@@ -63,9 +64,11 @@ def _sanitize_context(context):
         """
         if not context:
             return {}
-
+        client_id = context.get(Constants.CURRENT_CLIENT_FILTER_ID_PARAMTER_NAME)
         return {
-            Constants.ACTIVATE_BLOCK_ID_PARAMETER_NAME: context.get(Constants.ACTIVATE_BLOCK_ID_PARAMETER_NAME, None)
+            Constants.ACTIVATE_BLOCK_ID_PARAMETER_NAME: context.get(Constants.ACTIVATE_BLOCK_ID_PARAMETER_NAME, None),
+            Constants.CURRENT_CLIENT_FILTER_ID_PARAMTER_NAME:
+                None if client_id is None or not client_id.strip() else int(client_id)
```

Comment:
@jbzdak I would suggest doing this conversion outside of the dictionary (in an auxiliary variable), i.e.

   raw_client_id = context.get(Constants.CURRENT_CLIENT_FILTER_ID_PARAMTER_NAME)
   client_id = None
   if raw_client_id is not None and raw_client_id.strip():
        client_id = int(raw_client_id)